src/Interpreters/Context.cpp

```diff
@@ -293,6 +294,17 @@ struct ContextSharedPart
 
     ~ContextSharedPart()
     {
+        /// Wait for thread pool for background writes,
+        /// since it may use per-user MemoryTracker which will be destroyed here.
+        try
+        {
+            IDiskRemote::getThreadPoolWriter().wait();
+        }
+        catch (...)
+        {
+            tryLogCurrentException(__PRETTY_FUNCTION__);
+        }
+
         try
         {
             shutdown();
```

Comment:
Why not merge these 2 try/catch blocks?

Reply:
Since `shutdown` should be done even if `wait` will throw.

Reply:
Ah, sorry, that makes sense.